src/carcockpit/gltf_viewer_widget.cpp

```diff
@@ -0,0 +1,218 @@
+/*
+carcockpit - Car cockpit example GUI project
+
+Copyright (C) 2024 Gagistech Oy <gagisechoy@gmail.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+/* ================ LICENSE END ================ */
+
+#include "gltf_viewer_widget.hpp"
+
+#include <ratio>
+
+#include <GLES2/gl2.h>
+#include <GLES2/gl2ext.h>
+#include <GLES2/gl2platform.h>
+#include <papki/fs_file.hpp>
+#include <ruis/render/opengles/texture_2d.hpp>
+#include <ruis/res/texture_cube.hpp>
+
+#include "../ruis/render/scene/gltf_loader.hxx"
+
+#include "application.hpp"
+
+using namespace carcockpit;
+using namespace ruis::render;
+
+gltf_viewer_widget::gltf_viewer_widget(utki::shared_ref<ruis::context> context, all_parameters params) :
+	ruis::widget(std::move(context), {.widget_params = std::move(params.widget_params)}),
+	ruis::fraction_widget(this->context, {}),
+	params(std::move(params.gltf_params))
+{
+	LOG([&](auto& o) {
+		o << "[LOAD GLTF] " << this->params.path_to_gltf << std::endl;
+	})
+
+	ruis::render::gltf_loader l(*this->context.get().renderer.get().factory);
+
+	demoscene = l.load(papki::fs_file(this->params.path_to_gltf)).to_shared_ptr();
+
+	sc_renderer = std::make_shared<ruis::render::scene_renderer>(this->context);
+	sc_renderer->set_scene(demoscene);
+
+	camrip = std::make_shared<ruis::render::camera>();
+	sc_renderer->set_external_camera(camrip);
+	sc_renderer->set_scene_scaling_factor(this->params.scaling_factor);
+	sc_renderer->set_environment_cube(this->params.environment_cube);
+}
+
+void gltf_viewer_widget::update(uint32_t dt)
+{
+	demoscene->update(dt);
+
+	this->fps_sec_counter += dt;
+	this->time += dt;
+	[[maybe_unused]] float ft = static_cast<float>(this->time) / std::milli::den;
+	float fdt = static_cast<float>(dt) / std::milli::den;
+	++this->fps;
+
+	float xx = 3 * cosf(ft / 2);
+	float zz = 3 * sinf(ft / 2);
+
+	auto light = demoscene->get_primary_light();
+	if (light) {
+		light->pos = {xx, 3, zz, 1};
```

Comment:
move variables `xx` and `zz` declaration to this `if` block, those are not used anywhere else

Reply:
done